### PR TITLE
Fix and enhance wrong doc-strings of all `batch_transform` and `output_transform`

### PR DESCRIPTION
**Pull request by Nic-Ma**
### Description
According to Clara users' feedback and FAQs about handlers, this PR enhanced the doc-strings of all the `batch_transform` and `output_transform` args of handlers to explicitly indicate what this callable will **get** and from **where** and be used for **what**.
Also fixed some wrong doc-string and added missing doc-string.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/handlers/classification_saver.py [hunk 1]
```diff
@@ -57,12 +57,12 @@ def __init__(
             filename: if `saver=None`, name of the saved CSV file name.
             overwrite: if `saver=None`, whether to overwriting existing file content, if True,
                 will clear the file before saving. otherwise, will apend new content to the file.
-            batch_transform: a callable that is used to transform the
-                ignite.engine.batch into expected format to extract the meta_data dictionary.
-            output_transform: a callable that is used to transform the
-                ignite.engine.output into the form expected model prediction data.
-                The first dimension of this transform's output will be treated as the
-                batch dimension. Each item in the batch will be saved individually.
+            batch_transform: a callable that is used to extract the `meta_data` dictionary of
```
Comment (yiheng-wang-nv) [inline monai/handlers/classification_saver.py:60]: my suggestion is:
batch_transform: a callable that is used to transform the output batch achieved from `ignite.engine.state.batch` to extract the `meta_data` dictionary from the input images. The purpose of extracting the `meta_data` dictionary is to get the filenames of the input images in order to store the classification results.

File: monai/handlers/metrics_saver.py [hunk 2]
```diff
@@ -45,8 +45,9 @@ class MetricsSaver:
             "*" - save all the existing metric_details in `engine.state.metric_details` dict into separate files.
             list of strings - specify the metric_details of expected metrics to save.
             if not None, every metric_details array will save a separate `{metric name}_raw.csv` file.
-        batch_transform: callable function to extract the meta_dict from input batch data if saving metric details.
-            used to extract filenames from input dict data.
+        batch_transform: a callable that is used to extract the `meta_data` dictionary of
```
Comment (yiheng-wang-nv) [inline monai/handlers/metrics_saver.py:48]: my suggestion is:
batch_transform: a callable that is used to transform the output batch achieved from `ignite.engine.state.batch` to extract the `meta_data` dictionary from the input images. The purpose of extracting the `meta_data` dictionary is to get the filenames of the input images in order to store with metric details together.

File: monai/handlers/regression_metrics.py [hunk 3]
```diff
@@ -29,7 +29,8 @@ def __init__(
         """
 
         Args:
-            output_transform: transform the ignite.engine.state.output into [y_pred, y] pair.
+            output_transform: callable to construct the `[y_pred, y]` pair from `ignite.engine.state.output`, where
```
Comment (yiheng-wang-nv) [inline monai/handlers/regression_metrics.py:32]: Same suggestion as written in `ignite_metric.py`.

File: monai/handlers/roc_auc.py [hunk 4]
```diff
@@ -33,10 +33,8 @@ class ROCAUC(IgniteMetric):  # type: ignore[valid-type, misc]  # due to optional
                 indicator matrix as a label.
             - ``"none"``: the scores for each class are returned.
 
-        output_transform: a callable that is used to transform the
-            :class:`~ignite.engine.Engine` `process_function` output into the
-            form expected by the metric. This can be useful if, for example, you have a multi-output model and
-            you want to compute the metric with respect to one of the outputs.
+        output_transform: callable to construct the `[y_pred, y]` pair from `ignite.engine.state.output`,
```
Comment (yiheng-wang-nv) [inline monai/handlers/roc_auc.py:36]: Same suggestion as written in `ignite_metric.py`.

File: monai/handlers/segmentation_saver.py [hunk 5]
```diff
@@ -101,13 +101,12 @@ def __init__(
...
                 data_root_dir: /foo/bar,
                 output will be: /output/test1/image/image_seg.nii.gz
-            batch_transform: a callable that is used to transform the
-                ignite.engine.batch into expected format to extract the meta_data dictionary.
-                it can be used to extract the input image meta data: filename, affine, original_shape, etc.
-            output_transform: a callable that is used to transform the
-                ignite.engine.output into the form expected image data.
-                The first dimension of this transform's output will be treated as the
-                batch dimension. Each item in the batch will be saved individually.
+            batch_transform: a callable that is used to extract the `meta_data` dictionary of the input images
```
Comment (yiheng-wang-nv) [inline monai/handlers/segmentation_saver.py:104]: my suggestion is:
batch_transform: a callable that is used to transform the output batch achieved from `ignite.engine.state.batch` to extract the `meta_data` dictionary from the input images. The purpose of extracting the `meta_data` dictionary is to extract some necessary information such as filename, affine, original_shape, etc.

File: monai/handlers/segmentation_saver.py [hunk 6]
```diff
@@ -101,13 +101,12 @@ def __init__(
...
+                from `ignite.engine.batch`. then extract necessary information from the meta data: filename,
+                affine, original_shape, etc.
+            output_transform: a callable that is used to extract the model prediction data from
```
Comment (yiheng-wang-nv) [inline monai/handlers/segmentation_saver.py:107]: my suggestion is:
output_transform: a callable that is used to transform the segmentation predictions achieved from `ignite.engine.state.output`. The first dimension of its output will be treated as the batch dimension and each item in the batch will be saved individually.

File: monai/handlers/surface_distance.py [hunk 7]
```diff
@@ -38,7 +38,8 @@ def __init__(
                 `seg_pred` and `seg_gt`. Defaults to ``False``.
             distance_metric: : [``"euclidean"``, ``"chessboard"``, ``"taxicab"``]
                 the metric used to compute surface distance. Defaults to ``"euclidean"``.
-            output_transform: transform the ignite.engine.state.output into [y_pred, y] pair.
+            output_transform: callable to construct the `[y_pred, y]` pair from `ignite.engine.state.output`, where
```
Comment (yiheng-wang-nv) [inline monai/handlers/surface_distance.py:41]: Same suggestion as written in `ignite_metric.py`.

File: monai/handlers/tensorboard_handlers.py [hunk 8]
```diff
@@ -258,10 +258,10 @@ def __init__(
             interval: plot content from engine.state every N epochs or every N iterations, default is 1.
             epoch_level: plot content from engine.state every N epochs or N iterations. `True` is epoch level,
                 `False` is iteration level.
-            batch_transform: a callable that is used to transform the
-                ``ignite.engine.batch`` into expected format to extract several label data.
-            output_transform: a callable that is used to transform the
-                ``ignite.engine.output`` into expected format to extract several output data.
+            batch_transform: a callable that is used to construct `[images, labels]` pair from `engine.state.batch`.
```
Comment (yiheng-wang-nv) [inline monai/handlers/tensorboard_handlers.py:261]: my suggestion is:
batch_transform: a callable that is used to transform the output batch achieved from `ignite.engine.state.batch` to construct `[images, labels]` pair ...

File: monai/handlers/tensorboard_handlers.py [hunk 9]
```diff
@@ -258,10 +258,10 @@ def __init__(
...
+                then plot image from `transformed[0][index]` and plot label from `transformed[1][index]`.
+            output_transform: a callable that is used to extract the `predictions` data from `ignite.engine.output`,
```
Comment (yiheng-wang-nv) [inline monai/handlers/tensorboard_handlers.py:263]: output_transform: a callable that is used to transform the output achieved from `ignite.engine.state.output`...

File: monai/handlers/classification_saver.py [hunk 10]
```diff
@@ -57,12 +57,12 @@ def __init__(
...
+                the input images from `ignite.engine.batch`. then get the input filenames from
+                the `meta_data` to store classification results.
+            output_transform: a callable that is used to extract the model prediction data from
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/handlers/classification_saver.py:63]: Thanks for your suggestion. Yes, it should be `engine.state.output`.
But the `output_transform` is used to get the prediction from `engine.state.output`, not transform the prediction achieved from `engine.state.output`. For example, `engine.state.output` is `{"image": xxx, "pred": xxx}`, `output_transform` is `lambda x: x["pred"]`.

Thanks.

File: monai/handlers/ignite_metric.py [hunk 11]
```diff
@@ -35,7 +35,8 @@ class IgniteMetric(Metric):  # type: ignore[valid-type, misc] # due to optional_
     Args:
         metric_fn: callable function or class to compute raw metric results after every iteration.
             expect to return a Tensor with shape (batch, channel, ...) or tuple (Tensor, not_nans).
-        output_transform: transform the ignite.engine.state.output into [y_pred, y] pair.
+        output_transform: callable to construct the `[y_pred, y]` pair from `ignite.engine.state.output`, where
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/handlers/ignite_metric.py:38]: updated in the latest commit.
Thanks.

File: monai/handlers/confusion_matrix.py [hunk 12]
```diff
@@ -40,7 +40,8 @@ def __init__(
                 ``"informedness"``, ``"markedness"``]
                 Some of the metrics have multiple aliases (as shown in the wikipedia page aforementioned),
                 and you can also input those names instead.
-            output_transform: transform the ignite.engine.state.output into [y_pred, y] pair.
+            output_transform: callable to construct the `[y_pred, y]` pair from `ignite.engine.state.output`, where
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/handlers/confusion_matrix.py:43]: updated in the latest commit.
Thanks.

File: monai/handlers/hausdorff_distance.py [hunk 13]
```diff
@@ -41,7 +41,8 @@ def __init__(
                 percentile of the Hausdorff Distance rather than the maximum result will be achieved.
                 Defaults to ``None``.
             directed: whether to calculate directed Hausdorff distance. Defaults to ``False``.
-            output_transform: transform the ignite.engine.state.output into [y_pred, y] pair.
+            output_transform: callable to construct the `[y_pred, y]` pair from `ignite.engine.state.output`, where
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/handlers/hausdorff_distance.py:44]: updated in the latest commit.
Thanks.

File: monai/handlers/mean_dice.py [hunk 14]
```diff
@@ -32,7 +32,8 @@ def __init__(
         Args:
             include_background: whether to include dice computation on the first channel of the predicted output.
                 Defaults to True.
-            output_transform: transform the ignite.engine.state.output into [y_pred, y] pair.
+            output_transform: callable to construct the `[y_pred, y]` pair from `ignite.engine.state.output`, where
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/handlers/mean_dice.py:35]: updated in the latest commit.
Thanks.

File: monai/handlers/regression_metrics.py (hunk 3 above)
Comment (Nic-Ma) [reply monai/handlers/regression_metrics.py:32]: updated in the latest commit.
Thanks.

File: monai/handlers/roc_auc.py (hunk 4 above)
Comment (Nic-Ma) [reply monai/handlers/roc_auc.py:36]: updated in the latest commit.
Thanks.

File: monai/handlers/surface_distance.py (hunk 7 above)
Comment (Nic-Ma) [reply monai/handlers/surface_distance.py:41]: updated in the latest commit.
Thanks.

File: monai/handlers/classification_saver.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/handlers/classification_saver.py:60]: updated in the latest commit.
Thanks.

File: monai/handlers/metrics_saver.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/handlers/metrics_saver.py:48]: updated in the latest commit.
Thanks.

File: monai/handlers/segmentation_saver.py (hunk 5 above)
Comment (Nic-Ma) [reply monai/handlers/segmentation_saver.py:104]: updated in the latest commit.
Thanks.

File: monai/handlers/segmentation_saver.py (hunk 6 above)
Comment (Nic-Ma) [reply monai/handlers/segmentation_saver.py:107]: updated in the latest commit.
Thanks.

File: monai/handlers/tensorboard_handlers.py (hunk 8 above)
Comment (Nic-Ma) [reply monai/handlers/tensorboard_handlers.py:261]: updated in the latest commit.
Thanks.

File: monai/handlers/tensorboard_handlers.py (hunk 9 above)
Comment (Nic-Ma) [reply monai/handlers/tensorboard_handlers.py:263]: updated in the latest commit.
Thanks.

**Review by yiheng-wang-nv — APPROVED** [review]

**Pull request merged**